### ISSUE TITLE
Getting compile error when trying to cross compile using minigw32

### ISSUE THREAD

**Issue by abhayap** [PortAudio/portaudio#685]
I'm getting the following error when I try to cross compile using minigw32.

```
[05:46:49]          typedef /* [unique] */  __RPC_unique_pointer BYTE_BLOB *UP_BYTE_BLOB;
[05:46:49]                                                                  ^
[05:46:49] In file included from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/combaseapi.h:153:0,
[05:46:49]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/objbase.h:14,
[05:46:49]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/ole2.h:17,
[05:46:49]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypes.h:13,
[05:46:49]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/winscard.h:10,
[05:46:49]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/windows.h:97,
[05:46:49]                  from /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
[05:46:49] /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypesbase.h:322:20: note: previous declaration of ‘UP_BYTE_BLOB’ was here
[05:46:49]  typedef BYTE_BLOB *UP_BYTE_BLOB;
[05:46:49]                     ^
[05:46:49] make[2]: *** [CMakeFiles/portaudio_static.dir/build.make:350: CMakeFiles/portaudio_static.dir/src/hostapi/wasapi/pa_win_wasapi.c.obj] Error 1
[05:46:49] make[2]: Leaving directory '/workspace/srcdir/build'
[05:46:49] make[1]: *** [CMakeFiles/Makefile2:126: CMakeFiles/portaudio_static.dir/all] Error 2
[05:46:49] make[1]: Leaving directory '/workspace/srcdir/build'
[05:46:49] make: *** [Makefile:150: all] Error 2
[05:46:49]  ---> make
[05:46:49] Previous command exited with 2
[05:46:49] Child Process exited, exit code 2
ERROR: LoadError: Build for libportaudio on i686-w64-mingw32 did not complete successfully

Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] autobuild(dir::AbstractString, src_name::AbstractString, src_version::VersionNumber, sources::Vector{var"#s949"} where var"#s949"<:BinaryBuilderBase.AbstractSource, script::AbstractString, platforms::Vector{T} where T, products::Vector{var"#s948"} where var"#s948"<:Product, dependencies::Vector{var"#s947"} where var"#s947"<:BinaryBuilderBase.AbstractDependency; verbose::Bool, debug::Bool, skip_audit::Bool, ignore_audit_errors::Bool, autofix::Bool, code_dir::Union{Nothing, String}, require_license::Bool, kwargs::Any)
   @ BinaryBuilder /depot/packages/BinaryBuilder/8s2gr/src/AutoBuild.jl:798
 [3] build_tarballs(ARGS::Any, src_name::Any, src_version::Any, sources::Any, script::Any, platforms::Any, products::Any, dependencies::Any; julia_compat::String, kwargs::Any)
   @ BinaryBuilder /depot/packages/BinaryBuilder/8s2gr/src/AutoBuild.jl:319
 [4] top-level scope
   @ /agent/_work/1/s/L/libportaudio/build_tarballs.jl:66
in expression starting at /agent/_work/1/s/L/libportaudio/build_tarballs.jl:66
##[error]Bash exited with code '1'.
Finishing: run build_tarballs.jl
```

I'm trying to merge this PR and every other platform is working: https://github.com/JuliaPackaging/Yggdrasil/pull/4158

Thanks!

**Comment by giordano** [comment]
Note that the log reported above is incomplete:
```
[11:40:49] /opt/bin/i686-w64-mingw32-libgfortran3-cxx03/i686-w64-mingw32-gcc -DPAWIN_USE_WDMKS_DEVICE_INFO -DPA_LITTLE_ENDIAN -DPORTAUDIO_CMAKE_GENERATED -DWINVER=0x0501 -D_CRT_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0501 -DUNICODE -D_UNICODE @CMakeFiles/portaudio_static.dir/includes_C.rsp -g   -o CMakeFiles/portaudio_static.dir/src/hostapi/wasapi/pa_win_wasapi.c.obj   -c /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c
[11:40:50] /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:175:24: error: redefinition of ‘struct _BYTE_BLOB’
[11:40:50]          typedef struct _BYTE_BLOB
[11:40:50]                         ^
[11:40:50] In file included from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/combaseapi.h:153:0,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/objbase.h:14,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/ole2.h:17,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypes.h:13,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/winscard.h:10,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/windows.h:97,
[11:40:50]                  from /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
[11:40:50] /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypesbase.h:317:16: note: originally defined here
[11:40:50]  typedef struct _BYTE_BLOB {
[11:40:50]                 ^
[11:40:50] /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:179:15: error: conflicting types for ‘BYTE_BLOB’
[11:40:50]          }     BYTE_BLOB;
[11:40:50]                ^
[11:40:50] In file included from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/combaseapi.h:153:0,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/objbase.h:14,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/ole2.h:17,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypes.h:13,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/winscard.h:10,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/windows.h:97,
[11:40:50]                  from /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
[11:40:50] /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypesbase.h:320:3: note: previous declaration of ‘BYTE_BLOB’ was here
[11:40:50]  } BYTE_BLOB;
[11:40:50]    ^
[11:40:50] /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:180:65: error: conflicting types for ‘UP_BYTE_BLOB’
[11:40:50]          typedef /* [unique] */  __RPC_unique_pointer BYTE_BLOB *UP_BYTE_BLOB;
[11:40:50]                                                                  ^
[11:40:50] In file included from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/combaseapi.h:153:0,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/objbase.h:14,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/ole2.h:17,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypes.h:13,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/winscard.h:10,
[11:40:50]                  from /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/windows.h:97,
[11:40:50]                  from /workspace/srcdir/portaudio/src/hostapi/wasapi/pa_win_wasapi.c:46:
[11:40:50] /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/wtypesbase.h:322:20: note: previous declaration of ‘UP_BYTE_BLOB’ was here
[11:40:50]  typedef BYTE_BLOB *UP_BYTE_BLOB;
[11:40:50]                     ^
[11:40:50] make[2]: *** [CMakeFiles/portaudio_static.dir/build.make:350: CMakeFiles/portaudio_static.dir/src/hostapi/wasapi/pa_win_wasapi.c.obj] Error 1
```

**Comment by giordano** [comment]
This is a duplicate of #354, fixed by #552
